lib/Sema/DerivedConformanceEquatableHashable.cpp

```diff
@@ -1013,12 +1015,12 @@ deriveHashable_hashValue(TypeChecker &tc, Decl *parentDecl,
   //     case A:
   //       result = 0
   //     case B(let a0):
-  //       result = _mixForSynthesizedHashValue(result, 1)
-  //       result = _mixForSynthesizedHashValue(result, a0.hashValue)
+  //       result = _combineHashValues(result, 1)
+  //       result = _combineHashValues(result, a0.hashValue)
   //     case C(let a0, let a1):
-  //       result = _mixForSynthesizedHashValue(result, 2)
-  //       result = _mixForSynthesizedHashValue(result, a0.hashValue)
-  //       result = _mixForSynthesizedHashValue(result, a1.hashValue)
+  //       result = _combineHashValues(result, 2)
+  //       result = _combineHashValues(result, a0.hashValue)
+  //       result = _combineHashValues(result, a1.hashValue)
   //     }
   //     result = _mixInt(result)
```

Comment:
Do you think it would be worthwhile to eliminate these final calls to `_mixInt` now? They might be redundant since the seed in your new algorithm starts out well-randomized already.

Reply:
They might be redundant now, but I haven't thought it through and it can't hurt, so we can wait on that, maybe?

Reply:
That's fine—it might also be good (in a future PR) to have benchmarks against the synthesized `hashValue` so that future enhancements can be assured to not have performance regressions. (Not that I believe that would be the case here.)